src/udf/src/error.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use arrow_flight::error::FlightError;
+
+/// A specialized `Result` type for UDF operations.
+pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
nit: I find in this way we can make it more flexible:

https://github.com/dtolnay/anyhow/commit/fdd60501d7e93eab70fdb584588ef6199b1c61b3